core/async-derive/src/lib.rs

```diff
@@ -0,0 +1,418 @@
+use proc_macro::TokenStream;
+use proc_macro2::Span;
+use quote::quote;
+use syn::Meta;
```

Comment:
I'm missing some short explanation in the beginning of the file what problem does it solve, because it looks simple and helps to realise how useful is this. Rough example - 

"This helps to make any structure of type [struct ... { senders ... } ] implement trait allowing to send messages directly to this structure instead of calling each particular sender. Useful for Client and MockClient which have a lot of sub-modules exchanging messages [example]".

Probably there should be some limitations mentioned, like that there can be only one type for each message to send. We can also point a reader to tests after that because they are very demonstrative.